internal/internal_worker.go

```diff
@@ -1361,6 +1363,11 @@ func NewAggregatedWorker(client *WorkflowClient, taskQueue string, options Worke
 		DefaultHeartbeatThrottleInterval:      options.DefaultHeartbeatThrottleInterval,
 		MaxHeartbeatThrottleInterval:          options.MaxHeartbeatThrottleInterval,
 		cache:                                 cache,
+		eagerActivityExecutor: newEagerActivityExecutor(eagerActivityExecutorOptions{
+			disabled:      options.DisableEagerActivities,
+			taskQueue:     taskQueue,
+			maxConcurrent: options.MaxConcurrentEagerActivityExecutionSize,
```

Comment:
I'd take the min of MaxConcurrentEagerActivityExecutionSize and MaxConcurrentActivityExecutionSize to avoid unnecessarily reading from the channel

Reply:
You could also consider warning if eager max is greater than general max

Reply:
The channel is already bound by max concurrent activity execution size